ROADMAP.md

```diff
@@ -0,0 +1,58 @@
+# go-libp2p roadmap Q4’22/Q1’23
+
+## 📊 Comprehensive Metrics
+
+Why: For far too long, go-libp2p has been a black box. This has hurt us many times, by allowing trivial bugs to go undetected for a long time ([example](https://github.com/ipfs/kubo/pull/8750)). Having metrics will allow us to track the impact of performance improvements we make over time.
+
+Goal: go-libp2p ships with Grafana dashboards that allow node operators to monitor their nodes in production.
+
+How: This will look similar to how we already expose resource manager metrics. Metrics can be added incrementally for libp2p’s components. First milestone is having metrics for the swarm.
```

Comment:
Is the goal here to actually check in a Grafana dashboard json model to this repo and direct people to use that for monitoring? or only to slowly add metrics to go-libp2p components and enable users of go-libp2p to craft their own Grafana dashboards

Reply:
The goal is to have a ready-to-use Grafana dashboard _somewhere_. Of course, users are free to ignore our dashboards and build whatever they like from the metrics we expose.

Not sure if checking in the json file here is the most user-friendly way, especially as the number of json files will grow quickly. I don't think we need to figure out the best way to publish those right now, that can be part of the work we're planning to do.

Reply:
Yeah that makes sense. Other projects usually have community members that create their own dashboards for use by others. I'm thinking of something like this: https://github.com/dguess/mina-node-dashboard

> Not sure if checking in the json file here is the most user-friendly way

I agree, "go-libp2p ships with Grafana dashboards" just threw me off. Thanks for the clarification.

Reply:
Any suggestion how to rephrase?

Reply:
I added my suggestion https://github.com/libp2p/go-libp2p/pull/1784/files#r981794624

Reply:
Resolving this thread, since I've reworded the goal to be more specific.